### PR TITLE
Add configdb

### PR DESCRIPTION
**Pull request by jnation3406**
This adds an optional connection to configdb to get available configuration types and caches that. Will fallback on environment variable if no data is retrieved from configdb or no URL is set. The hardcoded list of configuration types in the model will be removed later when we remove the old field of OBSTYPE that uses it. The new field has been plumbed to use the dynamic list of configuration types already.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -15,6 +15,7 @@ uritemplate==3.0.1
 kombu>=4.6,<4.7
 ocs_archive
 setuptools-scm==6.0.1
+requests
```
Comment (mgdaily) [inline requirements.txt:18]: might be good to pin the version

File: archive/frames/utils.py [hunk 2]
```diff
@@ -32,6 +34,39 @@ def archived_queue_payload(dictionary, frame):
...
+        cache.set('configuration_type_tuples', configuration_type_tuples, 3600)
+
+    return configuration_type_tuples
+
+
+def get_configdb_data():
+    instrument_data = cache.get('configdb_instrument_data')
+    if not instrument_data:
+        if settings.CONFIGDB_URL:
+            url = settings.CONFIGDB_URL + '/instruments/'
```
Comment (mgdaily) [inline archive/frames/utils.py:58]: Not a big deal, but I like using `urllib.parse.urljoin` for constructing URLs. Tends to be safer.

https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin

File: archive/frames/utils.py [hunk 3]
```diff
@@ -32,6 +34,39 @@ def archived_queue_payload(dictionary, frame):
...
+            try:
+                response = requests.get(url)
+                response.raise_for_status()
+                instrument_data = response.json()['results']
+                cache.set('configdb_instrument_data', instrument_data, 3600)
+            except (requests.exceptions.RequestException, requests.exceptions.HTTPError) as e:
```
Comment (mgdaily) [inline archive/frames/utils.py:64]: Might not need the `as e:` part of this except clause - not being used or logged anywhere.

Now that I think about it, a log message wouldn't hurt.

**Review by mgdaily — COMMENTED** [review]
Mostly looks good! Just a couple of things. Did you plan on adding a test or two just as a sanity check?

**Comment by jnation3406** [comment]
> Mostly looks good! Just a couple of things. Did you plan on adding a test or two just as a sanity check?

Good idea. I thought it was too simple to warrant tests, but in writing the tests I found a bug so it was worth it!

**Pull request merged**